src/main/java/at/hannibal2/skyhanni/features/mining/glacitemineshaft/CorpseLOS.kt

```diff
@@ -0,0 +1,154 @@
+object CorpseLOS {
+    private val config get() = SkyHanniMod.feature.mining.corpseLOS
+
+    /**
+     * REGEX-TEST: §9Party §8> §bMVP§0+§b nobaboy§f: §rx: -164, y: 8, z: -154 | (Lapis Corpse)
+     * REGEX-TEST: Guild > §r§6♣ §b[MVP§0+§b] nobaboy§f: x: 141, y: 14, z: -131
+     * REGEX-TEST: §8[§r§6407§r§8] §r§6♣ §b[MVP§0+§b] nobaboy§f: x: -9, y: 135, z: 20 | (Tungsten Corpse)
+     */
+    private val corpseCoordsWaypoint by RepoPattern.pattern(
+        "corpse.waypoints",
+        "^(?:[^ ]+ > )?(?<playerName>.+)§f: (§r)?x: (?<x>-?\\d+),? y: (?<y>-?\\d+),? z: (?<z>-?\\d+)"
+    )
+
+    private val detectedArmorStands: MutableList<Corpse> = mutableListOf()
+    private val parsedLocations: MutableList<LorenzVec> = mutableListOf()
+
+    // Could move this into its own class to reduce clutter
+    enum class CorpseType(val helmetName: String, val color: LorenzColor) {
+        LAPIS("Lapis Armor Helmet", LorenzColor.DARK_BLUE),
+        UMBER("Yog Helmet", LorenzColor.GOLD),
+        TUNGSTEN("Mineral Helmet", LorenzColor.GRAY),
+        VANGUARD("Vanguard Helmet", LorenzColor.BLUE)
+    }
+
+    data class Corpse(val type: CorpseType, val coords: LorenzVec, var shared: Boolean = false)
+
+    private fun findCorpse() {
+        Minecraft.getMinecraft().theWorld ?: return
+        for (entity in EntityUtils.getAllEntities().filterIsInstance<EntityArmorStand>()) {
+            // 3 block distance because for some reason location didn't match when it should a couple of times
+            if (detectedArmorStands.any { it.coords.distance(entity.getLorenzVec()) <= 3}) continue
+            if (entity.showArms && entity.hasNoBasePlate() && !entity.isInvisible) {
+                for (offset in -1..3) { // Offset for search, feels a bit too borderline but best I can do to deal with glass/ice
+                    val canSee = entity.getLorenzVec().add(y = offset).canBeSeen()
+                    if (canSee) {
+                        val helmetDisplayName = StringUtils.stripControlCodes(entity.getCurrentArmor(3).displayName)
+                        val corpseType = CorpseType.entries.firstOrNull { it.helmetName == helmetDisplayName }
+                        corpseType?.let {
+                            detectedArmorStands.add(Corpse(type = it, coords = entity.getLorenzVec(), shared = false))
+                            sendCorpseAlert(it)
+                        }
+                        break // Break the inner loop if the corpse is detected from any offset
+                    }
+                }
+            }
+        }
+    }
+
+    private fun sendCorpseAlert(type: CorpseType) {
+        val message = "Located a ${type.name.firstLetterUppercase()} Corpse and marked its location with a waypoint."
+        ChatUtils.chat(message)
+    }
+
+    private fun shareCoords() {
+        for (corpse in detectedArmorStands) {
+            if (parsedLocations.any { corpse.coords.distance(it) <= 5}) corpse.shared = true
```

Comment:
Have to fix the detection of other players sending locations because it does not return